text/0000-handleCut.md

```diff
@@ -0,0 +1,51 @@
+# Summary
+
+Add a handleCut editor property to EditorView to allow for custom cut behavior.
+
+# Motivation
+
+The handleCut property would allow others to apply custom behavior to cut events while leveraging prosemirror's built in cross browser support for the clipboard api. One use case is for not deleting the selected content on cut but instead adding a mark to the content. For example: tracking changes made to a document by striking a line through deleted content instead of removing it.
```

Comment:
I believe that trying to track changes by overriding every possible way of deletion and replacing it with adding marks is not a good idea. And even if you wanted to do that, it'd be much better done at the `filterTransaction` level than at the UI level. (Not that that invalidates this whole RFC, just that this particular use case is probably a bad example.)

The *recommended* way to track changes is to store steps, use [prosemirror-changeset](https://github.com/prosemirror/prosemirror-changeset) to condense them to a set of insertions and deletions, and then use decorations to show those in the document.